ext/IntervalArithmeticArblibExt.jl

```diff
@@ -0,0 +1,153 @@
+module IntervalArithmeticArblibExt
+
+import IntervalArithmetic as IA
+import Arblib
+
+# Promotion rules
+
+# These are needed for ambiguity reasons
+Base.promote_rule(
+    ::Type{IA.Interval{T}},
+    ::Type{S},
+) where {T<:IA.NumTypes,S<:Arblib.ArfOrRef} = IA.Interval{promote_type(T, S)}
+Base.promote_rule(
+    ::Type{T},
+    ::Type{IA.Interval{S}},
+) where {T<:Arblib.ArfOrRef,S<:IA.NumTypes} = IA.Interval{promote_type(T, S)}
+
+Base.promote_rule(
+    ::Type{IA.Interval{T}},
+    ::Type{S},
+) where {T<:IA.NumTypes,S<:Arblib.ArbOrRef} =
+    throw(ArgumentError("promotion between Interval and Arb is purposely not supported"))
+Base.promote_rule(
+    ::Type{T},
+    ::Type{IA.Interval{S}},
+) where {T<:Arblib.ArbOrRef,S<:IA.NumTypes} =
+    throw(ArgumentError("promotion between Interval and Arb is purposely not supported"))
+
+# Construction of Arb from Interval
+# Note that this automatically gives support for construction of Acb.
+
+Arblib._precision(x::Union{IA.Interval,IA.BareInterval}) =
+    Arblib._precision(IA.inf(x), IA.sup(x))
+
+function Arblib.set!(
+    res::Arblib.ArbLike,
+    x::Union{IA.Interval,IA.BareInterval};
+    prec::Integer = precision(res),
+)
+    if IA.isempty_interval(x)
+        Arblib.indeterminate!(res)
+    else
+        Arblib.set!(res, (IA.inf(x), IA.sup(x)); prec)
+    end
+    return res
+end
+
+# Construction of Interval from Arb
+
+# We essentially treat Arb as Interval{BigFloat} for the purposes of
+# constructing Intervals. This means that we only have to change the
+# behavior of _inf and _sup since these are called early on in the
+# construction process to handle endpoints which are Intervals.
+
+IA._inf(x::Arblib.ArbOrRef) = BigFloat(Arblib.lbound(x))
+IA._sup(x::Arblib.ArbOrRef) = BigFloat(Arblib.ubound(x))
+
+# Make promote_numtype for Arb behave like promotion with BigFloat
+IA.promote_numtype(::Type{<:Arblib.ArbOrRef}, ::Type{<:Arblib.ArbOrRef}) = BigFloat
+IA.promote_numtype(::Type{<:Arblib.ArbOrRef}, ::Type{S}) where {S} =
+    IA.promote_numtype(BigFloat, S)
+IA.promote_numtype(::Type{<:Arblib.ArbOrRef}, ::Type{S}) where {S<:IA.NumTypes} =
+    IA.promote_numtype(BigFloat, S)
+IA.promote_numtype(::Type{T}, ::Type{<:Arblib.ArbOrRef}) where {T} =
+    IA.promote_numtype(T, BigFloat)
+IA.promote_numtype(::Type{T}, ::Type{<:Arblib.ArbOrRef}) where {T<:IA.NumTypes} =
+    IA.promote_numtype(T, BigFloat)
+
+# Construction of Complex{Interval} from Acb
+IA.numtype(::Type{<:Arblib.AcbOrRef}) = Arblib.Arb
+
+# Handle Acb in the same way as Complex.
+IA._interval_infsup(
+    ::Type{T},
+    a::Arblib.AcbOrRef,
+    b::Arblib.AcbOrRef,
+    d::IA.Decoration = com,
```

Comment:
Is a default value for the decoration really needed? And if so, shouldn't that be `IA.com`?

Reply:
Good point, it should be `IA.com`. Indeed, it is probably not needed. I copied it from the [here](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/d948dabc770de047b8a74b628b5c4238af6d5026/src/intervals/construction.jl#L462C1-L474C1) and I see now that the these have default values for `d` but the versions above don't. I'll remove the default value.